src/utils/statsCheckings.js

```diff
@@ -0,0 +1,58 @@
+import Response from './ResponseHandler';
+/**
+ *
+ * @description view stats class
+ * @class StatsController
+ */
+export default class StatsController {
+  /**
+   * @param {object} allTrips
+   * @param {object} res
+   * @return {object} return stats
+   */
+  static async stats(allTrips, res) {
+    const createdTrips = allTrips.length;
+    const currentYear = new Date().getFullYear();
+    const currentMonth = new Date().getMonth();
+    let lastMonth, openRequests, approvedRequests, rejectedRequets, previousMonths, oldTrips;
+    const newArry = [], previousMonthsArray = [],
+      previousYearsArray = [], openRequestArr = [],
+      approvedRequestArr = [], rejectedRequetsArr = [];
+    allTrips.forEach(element => {
+      switch (element.status) {
+        case 'open':
+          openRequestArr.push(element);
+          openRequests = openRequestArr.length;
+          break;
+        case 'approved':
+          approvedRequestArr.push(element);
+          approvedRequests = approvedRequestArr.length;
+          break;
+        case 'rejected':
+          rejectedRequetsArr.push(element);
+          rejectedRequets = rejectedRequetsArr.length;
+          break;
+        default:
+      }
+      if (element.createdAt.getFullYear() === currentYear) {
+        if (element.createdAt.getMonth() < (currentMonth - 1)) {
+          newArry.push(element);
+          lastMonth = newArry.length;
```

Comment:
hey @bbaime98 won't this if statement find request made in the previous months rather than last month

Reply:
I implemented it in a way that users can see recent stats which are of last month and also the ones of previous months.

Reply:
but that would be a good idea of giving general stats of previous months